### PR TITLE
feat: Add export of saved data browser filters via `classPreference` settings

### LINKED ISSUE #2454: Allow classPreferences to be set in dashboard config
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Current Limitation

New feature views need to be added on every device, there is no way to keep filters persistent.

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description

Currently, filters have to be created every time and there's no way to define / export filters.

Closes: #2454

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Review by mtrezza — COMMENTED** [review]
Where would that feature be documented, so that users know it exists and how to use it?

**Comment by dblythy** [comment]
Added docs

File: README.md [hunk 1]
```diff
@@ -329,13 +329,33 @@ If you have classes with a lot of columns and you filter them often with the sam
...
+          }
         ]
       }
     }
 ]
 ```
 
+### Persistent Filters
+
+`classPreference` allows for filters to be saved across sessions. To save the current filters, navigate to App Settings and click "Export Class Preferences".
```
Comment (mtrezza) [inline README.md]:
```suggestion
`classPreference` allows for filters to be saved across sessions. To save the current filters, navigate to *App Settings > Export Class Preferences*.

For example:
```

File: README.md [hunk 2]
```diff
@@ -329,13 +329,35 @@ If you have classes with a lot of columns and you filter them often with the sam
...
+          }
         ]
       }
     }
 ]
 ```
 
+### Persistent Filters
+
+`classPreference` allows for filters to be saved across sessions. To save the current filters, navigate to *App Settings > Export Class Preferences*.
```
Comment (mtrezza) [inline README.md]:
```suggestion
The `classPreference` setting allows for data browser filters to be saved across sessions. To save the current filters, navigate to *App Settings > Export Class Preferences*.
```

File: README.md [hunk 3]
```diff
@@ -329,13 +329,35 @@ If you have classes with a lot of columns and you filter them often with the sam
...
+          "filters": [
+            {
+              "name": "Filter Name",
+              "filter": "[{\"field\":\"objectId\",\"constraint\":\"exists\"}]"
+            }
+          ]
+        }
+      }
+]
+```
```
Comment (mtrezza) [inline README.md]:
````suggestion
```json
"apps": [{
  "classPreference": {
    "_Role": {
      "filters": [{
        "name": "Filter Name",
        "filter": "[{\"field\":\"objectId\",\"constraint\":\"exists\"}]"
      }]
    }
  }
}]
```
````

File: README.md [hunk 4]
```diff
@@ -329,13 +329,32 @@ If you have classes with a lot of columns and you filter them often with the sam
...
+
+For example:
+
+```json
+"apps": [{
+  "classPreference": {
+    "_Role": {
+      "filters": [{
+        "name": "Filter Name",
+        "filter": "[{\"field\":\"objectId\",\"constraint\":\"exists\"}]"
```
Comment (mtrezza) [inline README.md]: About the filter value, why does its value need to be escaped? Why couldn't we just pass an object that would make it also more readable.

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -452,7 +476,7 @@ With MFA enabled, a user must provide a one-time password that is typically boun
 
 The user requires an authenticator app to generate the one-time password. These apps are provided by many 3rd parties and mostly for free.
 
-If you create a new user by running `parse-dashboard --createUser`, you will be  asked whether you want to enable MFA for the new user. To enable MFA for an existing user, 
+If you create a new user by running `parse-dashboard --createUser`, you will be  asked whether you want to enable MFA for the new user. To enable MFA for an existing user,
 run `parse-dashboard --createMFA` to generate a `mfa` secret that you then add to the existing user configuration, for example:
```
Comment (mtrezza) [inline README.md]:
```suggestion
If you create a new user by running `parse-dashboard --createUser`, you will be  asked whether you want to enable MFA for the new user. To enable MFA for an existing user, run `parse-dashboard --createMFA` to generate a `mfa` secret that you then add to the existing user configuration, for example:
```

**Review by mtrezza — DISMISSED** [review]
Looks good!

File: README.md [hunk 6]
```diff
@@ -329,13 +329,37 @@ If you have classes with a lot of columns and you filter them often with the sam
...
+          }
         ]
       }
     }
 ]
 ```
 
+### Persistent Filters
+
+The `classPreference` setting allows for data browser filters to be saved across sessions. To save the current filters, navigate to *App Settings > Export Class Preferences*.
```
Comment (mtrezza) [inline README.md]:
```suggestion
The filters you save in the data browser of Parse Dashboard are only available for the current dashboard user in the current browser session. To make filters permanently available for all dashboard users of an app, you can define filters in the `classPreference` setting.
```

File: README.md [hunk 7]
```diff
@@ -329,13 +329,37 @@ If you have classes with a lot of columns and you filter them often with the sam
...
+            "field": "objectId",
+            "constraint": "exists"
+          }
+        ]
+      }]
+    }
+  }
+}]
+```
+
```
Comment (mtrezza) [inline README.md:362]:
```suggestion

You can conveniently create a filter definition without having to write it by hand by first saving a filter in the data browser, then exporting the filter definition under *App Settings > Export Class Preferences*.

```

**Review by mtrezza — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-alpha.13](https://github.com/parse-community/parse-dashboard/releases/tag/5.2.0-alpha.13)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-beta.2](https://github.com/parse-community/parse-dashboard/releases/tag/5.2.0-beta.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0](https://github.com/parse-community/parse-dashboard/releases/tag/5.2.0)

**Pull request merged**